log_lists.go

```diff
@@ -0,0 +1,89 @@
+package datadog
+
+import (
+	"time"
+)
+
+const logsListPath = "/v1/logs-queries/list"
+
+// LogsListRequest represents the request body sent to the list API
+type LogsListRequest struct {
+	Index   *string    `json:"index,omitempty"`
+	Limit   *int       `json:"limit,omitempty"`
+	Query   *string    `json:"query"`
+	Sort    *string    `json:"sort,omitempty"`
+	StartAt *string    `json:"startAt,omitempty"`
+	Time    *QueryTime `json:"time"`
+}
+
+// QueryTime represents the time object for the request sent to the list API
+type QueryTime struct {
+	TimeFrom *string `json:"from"`
+	TimeTo   *string `json:"to"`
+	TimeZone *string `json:"timezone,omitempty"`
+}
+
+// LogsList represents the base API response returned by the list API
+type LogsList struct {
+	Logs      []Logs  `json:"logs"`
+	NextLogID *string `json:"nextLogId"`
+	Status    *string `json:"status"`
+}
+
+func (l *LogsList) next() bool {
+	if l.NextLogID != nil || l.Status == nil {
```

Comment:
The status check handles the case where the `*LogsList` object is newly instantiated; `l.NextLogID`will be nil because we haven't made any requests, but the API should return a status after *any* request, whether successful or not. Basically, it indicates that no requests have been made at all since the object was instantiated.

Reply:
Disregard this, someone hadn't had her coffee yet yesterday. 🤦🏻‍♀️